lightning/src/util/ser.rs

```diff
@@ -381,6 +383,39 @@ impl Readable for BigSize {
 	}
 }
 
+/// The lightning protocol uses u16s for lengths in most cases. As our serialization framework
+/// primarily targets that, we must as well. However, because we may serialize objects that have
+/// more than 65K entries, we need to be able to store larger values. Thus, we define a variable
+/// length integer here which is backwards-compatible but treats 0xffff as "read eight more bytes".
+///
+/// To ensure we only have one valid encoding per value, we add 0xffff to values written as eight
+/// bytes. Thus, 0xfffe is serialized as 0xfffe, whereas 0xffff is serialized as
+/// 0xffff0000000000000000 (i.e. read-eight-bytes then zero).
```

Comment:
this mixed endianness is so confusing (0xfffe being big endian, but 0xffff0000000000000000 being little endian)

Reply:
How can the second number be big endian?

Reply:
oh, I didn't realize that the 0xffff portion is completely discarded, sorry.

Reply:
so we don't wanna do anything clever where we could encode up to `u64::max-0xffff`, and keep a total max encoding length of u64?

Reply:
yeah, probably not worth it, disregard this entire comment chain.